CHANGELOG.md

```diff
@@ -16,6 +16,12 @@ The following emojis are used to highlight certain changes:
 
 ### Added
 
+- `bitswap/client`: Improved timeout configuration for block requests
+  - Exposed `DontHaveTimeoutConfig` to hold configuration values for `dontHaveTimeoutMgr` which controls how long to wait for requested block, and if timeout should emit a synthetic DontHave response
+  - Added `DefaultDontHaveTimeoutConfig()` to return a `DontHaveTimeoutConfig` populated with default values
+  - Added optional `WithDontHaveTimeoutConfig` to allow passing a custom `DontHaveTimeoutConfig`
+  - If `DontHaveTimeoutConfig.onDontHaveTimeout` is `nil`, then `dontaHaveTimeoutMgr` is disabled and synthetic DontHave won't be emitted on timeout
```

Comment:
@lidel From a user's perspective this last point did not change. The only change was internal - to not run the timeout manager, instead of running it but calling a no-op function.

Reply:
@gammazero  Indeed, I was very confused what is the point of `DontHaveTimeoutConfig.onDontHaveTimeout`, after closer inspection it is not used for anything(?)

The one passed to `newDontHaveTimeoutMgr` as a separate argument is used instead, and we create it in bitswap client.

Removed `DontHaveTimeoutConfig.onDontHaveTimeout` in 068cd7b42e95b50e048f3a7a27e2e8dc8b0bed5f, now the public struct is just timeouts. 

Updated this line in changelog too. 
If there is not more to it, lgtm.